apps/jubmoji-quest/src/pages/api/quests.ts

```diff
@@ -0,0 +1,117 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import prisma from "@/lib/prisma";
+import { getRandomNullifierRandomness } from "jubmoji-api";
+import { bigIntToHex } from "babyjubjub-ecdsa";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "GET") {
+    try {
+      const quests = await prisma.quest.findMany({
+        select: {
+          id: true,
+          name: true,
+          description: true,
+          tags: true,
+          startTime: true,
+          endTime: true,
+          proofType: true,
+          proofParams: true,
+          // Selecting specific fields from related models
+          prerequisiteCards: {
+            select: {
+              index: true,
+            },
+          },
+          collectionCards: {
+            select: {
+              index: true,
+            },
+          },
+        },
+      });
+
+      const transformedQuests = quests.map((quest) => {
+        const prerequisiteIndices = quest.prerequisiteCards.map(
+          (card) => card.index
+        );
+        const collectionIndices = quest.collectionCards.map(
+          (card) => card.index
+        );
+        return {
+          ...quest,
+          prerequisiteCardIndices: prerequisiteIndices,
+          collectionCardIndices: collectionIndices,
+          // Remove the original fields from the response object
+          prerequisiteCards: undefined,
+          collectionCards: undefined,
+        };
+      });
+      res.status(200).json(transformedQuests);
+    } catch (error) {
+      res.status(500).json({ error: "Unable to fetch quests" });
+    }
+  } else if (req.method === "POST") {
+    try {
+      // Here, we're assuming the body contains the quest details.
+      // You should validate these details before attempting to create a quest in your database.
+
+      const {
+        name,
+        description,
+        tags,
+        startTime,
+        endTime,
+        proofType,
+        N,
+        prerequisiteCardIndices,
+        collectionCardIndices,
+      } = req.body;
+
+      // Todo: Add random nullifiers
```

Comment:
Can these sorts of todos be tracked somewhere in issues? Don't want to forget about cleaning this function up

Reply:
Unless you're just 100% sure you'll get to it during another task, but maybe it'd be good to have in-code documentation in that case

Reply:
This is a good point. Normally I keep todo's out of the codebase, but in the spirit of moving fast, we can leave quest/power creation as incomplete. I've created another ticket for either making quest/power creation a supported feature or deleting it: https://github.com/project-vinyl/jubmoji.quest/issues/41